src/mito2/src/read/unordered_scan.rs

```diff
@@ -135,6 +137,51 @@ impl UnorderedScan {
         }
     }
 
+    /// Scans a [PartitionRange] by its `identifier` and returns a flat stream of RecordBatch.
+    fn scan_flat_partition_range(
+        stream_ctx: Arc<StreamContext>,
+        part_range_id: usize,
+        part_metrics: PartitionMetrics,
+        range_builder_list: Arc<RangeBuilderList>,
+    ) -> impl Stream<Item = Result<RecordBatch>> {
+        try_stream! {
+            // Gets range meta.
+            let range_meta = &stream_ctx.ranges[part_range_id];
+            for index in &range_meta.row_group_indices {
+                if stream_ctx.is_mem_range_index(*index) {
+                    let stream = scan_flat_mem_ranges(
+                        stream_ctx.clone(),
+                        part_metrics.clone(),
+                        *index,
+                    );
+                    for await record_batch in stream {
+                        yield record_batch?;
+                    }
+                } else if stream_ctx.is_file_range_index(*index) {
+                    let stream = scan_flat_file_ranges(
+                        stream_ctx.clone(),
+                        part_metrics.clone(),
+                        *index,
+                        "unordered_scan_files",
+                        range_builder_list.clone(),
+                    ).await?;
+                    for await record_batch in stream {
+                        yield record_batch?;
+                    }
+                } else {
+                    let stream = scan_util::maybe_scan_flat_other_ranges(
```

Comment:
Do we still need to keep `maybe_scan_flat_other_ranges` function?

Reply:
We should implement this later